### PR TITLE
checkpoint2

### PR DESCRIPTION
**Pull request by SeanPhilippi**
## Checkpoint Rubric
This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

**Checkpoint 1**
- [ ] All tests passed: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 30 points

**Checkpoint 2**
- [ ] The application works as it should: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript and the DOM API: 30 points

**Checkpoint 3**
- [ ] Use of React: 25 points
- [ ] Accesses an API: 25 points
- [ ] Proper use of documentation (commenting on code): 25 points
- [ ] The application functions as it should: 25 points

### CODE REVIEW

File: 06week/checkpoint2.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    humidity: 52,
+    visibility: 11.14056410562316,
+    predictability: 71
+  }
+]
+
+//using a higher order function, create an array of the unique 'weather_state_name' values of the weather array.
+//Your function should return the following array ['Light Cloud', 'Heavy Cloud', 'Showers']
+
+const weatherStates = weather.map((arr) => arr['weather_state_name']);
```
Comment (wc0sby) [inline 06week/checkpoint2.js]: This should return an array of unique weathe state values.  There needs to be one more tweak to the syntax to make it return a unique array

**Review by wc0sby — COMMENTED** [review]
Good job, just one little modification is required to yield the expected return on the weatherStates variable

**Comment by SeanPhilippi** [comment]
const uniqueStates = weather.map((arr) => arr['weather_state_name'])
.filter((value, index, self) => self.indexOf(value) === index)

console.log(uniqueStates);

// how is this for a solution? Is there an easier way?

**Comment by reneemeyer** [comment]
@SeanPhilippi that solution works great!

**Pull request merged**